### PR TITLE
test: add additional scheduler integration tests for property-based scheduling (1/)

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR adds a few test cases for property-based scheduling in the scheduler integration tests. More PRs to ensue.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### Special notes for your reviewer

Also includes a minor fix (see #853).

### CODE REVIEW

**Comment by ryanzhang-oss** [comment]
can you please resolve the conflicts?

File: test/scheduler/property_based_scheduling_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+			ensureCRPAndAllRelatedResourcesDeletion(crpName)
+		})
+	})
+
+	Context("property selector updated", Ordered, func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+		policySnapshotName1 := fmt.Sprintf(policySnapshotNameTemplate, crpName, 1)
+		policySnapshotName2 := fmt.Sprintf(policySnapshotNameTemplate, crpName, 2)
+
+		wantScheduledClusters1 := []string{
```
Comment (zhiying-lin) [inline test/scheduler/property_based_scheduling_integration_test.go:432]: ditto

**Review by zhiying-lin — COMMENTED** [review]
two minor comments, others LGTM

File: test/scheduler/property_based_scheduling_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,834 @@
...
+		})
+	})
+
+	// This spec has been marked as serial as it updates the cluster properties, which may
+	// interfere with other specs if run in parallel.
+	Context("cluster properties refreshed", Serial, Ordered, func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+		policySnapshotName := fmt.Sprintf(policySnapshotNameTemplate, crpName, 1)
+
+		wantTargetClusters1 := []string{
```
[the remark this answers is not in the document]
Comment (michaelawyu) [reply test/scheduler/property_based_scheduling_integration_test.go:698]: Done.

File: test/scheduler/property_based_scheduling_integration_test.go (hunk 1 above)
Comment (michaelawyu) [reply test/scheduler/property_based_scheduling_integration_test.go:435]: Done.

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**